lib/galaxy/datatypes/metadata.py

```diff
@@ -763,8 +762,8 @@ def __get_filename_override():
             # so we will only populate the dictionary once
             metadata_files = self.get_output_filenames_by_dataset( dataset, sa_session )
             if not metadata_files:
-                metadata_files = galaxy.model.JobExternalOutputMetadata( dataset=dataset)
-                metadata_files.job_id = self.job_id
+                job = sa_session.query( galaxy.model.Job ).get( self.job_id )
+                metadata_files = galaxy.model.JobExternalOutputMetadata( job=job, dataset=dataset )
```

Comment:
This happens on the remote server right - my assumption would be that it should be able to work without a actually needing to query the database, am I wrong? I feel like I understand how that would work with the old lines but not with the new ones.